### PR TITLE
Day 2/sale

### PR DESCRIPTION
**Pull request by salac-kolosek**

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+const Button = ({className, isDisabled, imgSrc, onClick}) => (
+  <button 
+    className={className} 
+    disabled={isDisabled} 
+    onClick={onClick}
+  >
+    <img src={imgSrc} width="30" height="30" />
```
Comment (kristina-kolosek) [inline src/components/Button.js]: use css for defining width/height

File: src/components/Input.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+const Input = ({...props}) => (
```
Comment (kristina-kolosek) [inline src/components/Input.js:3]: no need for using spread operator here, you can do this:

const Input = (props) =>

File: src/components/TextArea.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+const TextArea = ({...props}) => (
```
Comment (kristina-kolosek) [inline src/components/TextArea.js:3]: Same as previous comment

File: src/containers/LeftPanel.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      <div className="left_panel-container">
+        {this.renderTopContent()}
+        {this.renderList()}
+      </div>
+    );
+  }
+
+  renderList() {
+    const { documents, selected } = this.props
+    let filteredDocuments = this.state.searchText.length < 2 ? documents :
```
Comment (kristina-kolosek) [inline src/containers/LeftPanel.js]: Remove minimum number of characters needed for search

File: src/containers/LeftPanel.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      filteredDocuments = documents.filter((doc) => {
+        return doc.title.includes(this.state.searchText)
```
Comment (kristina-kolosek) [inline src/containers/LeftPanel.js]: add checking if queried text is appearing in document content  as well, and ignore the casing when searching (ignore casing for title as well)

File: src/containers/LeftPanel.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      })
+    return(
+      <div className="left_panel-list">
+        {
+         filteredDocuments.map(doc => {
+            return(
+              <div 
+                key={doc.id} 
+                className={`left_panel-list__document-title ${doc.id === selected.id ? "left_panel-list__document-title--selected" : ""}`}
```
Comment (kristina-kolosek) [inline src/containers/LeftPanel.js:51]: Better define selectedClass variable above the return statement, if inline like this it's not so readable

File: src/containers/LeftPanel.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+      <div className="left_panel-list">
+        {
+         filteredDocuments.map(doc => {
+            return(
+              <div 
+                key={doc.id} 
+                className={`left_panel-list__document-title ${doc.id === selected.id ? "left_panel-list__document-title--selected" : ""}`} 
+                onClick={() => this.props.selectDocument(doc)}
+              >
+                {doc.title}
```
Comment (kristina-kolosek) [inline src/containers/LeftPanel.js:54]: Show identifier on whether the document was updated in editor or not

File: src/containers/MainPanel.js [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      text: '',
+    };
+  }
+
+  changeTitleInput = (event) => {
+    this.props.changeTitleInput(event.target.value)
```
Comment (kristina-kolosek) [inline src/containers/MainPanel.js]: is it possible to make changeTitle and changeText input functions to be 1 function? (both for action and a function)

File: src/containers/MainPanel.js [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+
+  changeTextInput = (event) => {
+     this.props.changeTextInput(event.target.value)
+  }
+
+  handleSave = () => {
+    if (this.props.titleInput.length > 1) {
+      if(this.props.mode === 'add') {
+        this.props.createDocument({
+          id: this.props.documents.length,
```
Comment (kristina-kolosek) [inline src/containers/MainPanel.js]: don't use length for id, generate id somehow (you can use timestamp for example)

File: src/containers/MainPanel.js [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+import SaveIcon from '../assets/images/pencil.png'
+
+import './css/Form.css'
+
+class MainPanel extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
```
Comment (kristina-kolosek) [inline src/containers/MainPanel.js]: Where do you use this state?

File: src/containers/MainPanel.js [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+    this.props.removeDocument(this.props.selected.id)
+  }
+
+  render() {
+    const { selected, mode, titleInput, textInput } = this.props
+    return (
+      <div className="document_editor-form">
+        { !isEmpty(selected) || mode === 'add'  ? 
+          <div className="editor_form-content">
+            {this.rederToolbar()}
```
Comment (kristina-kolosek) [inline src/containers/MainPanel.js]: nitpicking: spelling mistake

File: src/reducers/documents.js [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+    type: TEXT_INPUT_CHANGE,
+    value
+  }
+}
+
+let newList
+
+export default createReducer(initialState, ({
+  [CREATE_DOCUMENT]: (payload, state) => {
+    newList = state.list
```
Comment (kristina-kolosek) [inline src/reducers/documents.js]: don't mutate the state - you can use spread operator instead:

newList = [...state.list]

File: src/reducers/documents.js [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+      }
+    })
+    return {
+      ...state,
+      list: newList
+    }
+  },
+  [REMOVE_DOCUMENT]: (payload, state) => {
+    newList = state.list
+    newList = newList.filter(doc => doc.id !== payload.documentId)
```
Comment (kristina-kolosek) [inline src/reducers/documents.js]: This is unnecessary:

newList = state.list

just use state.list.filter straight away

**Review by kristina-kolosek — CHANGES_REQUESTED** [review]

File: src/components/Input.js (hunk 2 above)
Comment (salac-kolosek) [reply src/components/Input.js:3]: It doesn't work when I change it to just props.

File: src/components/TextArea.js (hunk 3 above)
Comment (salac-kolosek) [reply src/components/TextArea.js:3]: It doesn't work when I change it to just props.

File: src/components/Input.js (hunk 2 above)
Comment (kristina-kolosek) [reply src/components/Input.js:3]: Then you're not doing something right, probably mutating props somewhere

File: src/containers/LeftPanel.js [hunk 14]
```diff
@@ -36,9 +36,9 @@ class LeftPanel extends Component {
 
   renderList() {
     const { documents, selected } = this.props
-    let filteredDocuments = this.state.searchText.length < 2 ? documents :
-      filteredDocuments = documents.filter((doc) => {
-        return doc.title.includes(this.state.searchText)
+    let filteredDocuments = this.state.searchText.length === '' ? documents :
+      documents.filter((doc) => {
+        return doc.title.toLowerCase().includes(this.state.searchText.toLowerCase())
```
Comment (kristina-kolosek) [inline src/containers/LeftPanel.js]: Filter should be done both by title and document content

File: src/containers/LeftPanel.js [hunk 15]
```diff
@@ -36,9 +36,9 @@ class LeftPanel extends Component {
 
   renderList() {
     const { documents, selected } = this.props
-    let filteredDocuments = this.state.searchText.length < 2 ? documents :
-      filteredDocuments = documents.filter((doc) => {
-        return doc.title.includes(this.state.searchText)
+    let filteredDocuments = this.state.searchText.length === '' ? documents :
```
Comment (kristina-kolosek) [inline src/containers/LeftPanel.js]: You're comparing number to empty string here (although both '' and 0 are falsey, you should still care about types)

**Review by kristina-kolosek — CHANGES_REQUESTED** [review]
Besides the comments I left regarding code, there are some functionality issues:

* When adding a new document, if there was previously one selected, it still stays highlighted
* There's still no check if document was updated yet not saved when selecting different one

File: src/containers/LeftPanel.js (hunk 15 above)
Comment (salac-kolosek) [reply src/containers/LeftPanel.js]: I agree :)

Comment (salac-kolosek) [reply src/containers/LeftPanel.js]: Hm, it doesn't work when just this.state.searchText.length ?

**Pull request closed without merging**